profanity.rb

```diff
@@ -1004,7 +1004,8 @@ def get_color_pair_id(fg_code, bg_code)
 	# fixme: gsub %whatever
 	backslash = false
 	at_pos = nil
-	macro.split('').each { |ch|
+	backfill = nil
+	macro.split('').each_with_index { |ch, i|
```

Comment:
Definitely understand and I'm usually a "When in Rome..." guy, but I think this project would benefit with a push towards more idiomatic Ruby which is the main reason I suggested it here. This is a minor nit and I'm not aware of any major downsides from doing it this way, so this is not a blocker by any means.